From 2c6d48906078629f50f3dfca169a660e315de0a7 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 265/296] BackupAgent: add backup shared mode

default backup agent didn't support -shared mode backup.
add to support user storage backup.
---
 core/java/android/app/backup/BackupAgent.java | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/core/java/android/app/backup/BackupAgent.java b/core/java/android/app/backup/BackupAgent.java
index 7aa80d26397..1d56e068c16 100644
--- a/core/java/android/app/backup/BackupAgent.java
+++ b/core/java/android/app/backup/BackupAgent.java
@@ -648,8 +648,11 @@ public abstract class BackupAgent extends ContextWrapper {
         // Pull out the domain and set it aside to use when making the tarball.
         String domainPath = FullBackup.getBackupScheme(this).tokenToDirectoryPath(domain);
         if (domainPath == null) {
-            // Should never happen.
-            return;
+            if (startingPath == null) {
+                return;
+            } else {
+                domainPath = startingPath;
+            }
         }
 
         File rootFile = new File(startingPath);
-- 
2.11.0

